sql/mysqld.cc

```diff
@@ -2279,6 +2280,14 @@ void clean_up(bool print_message)
   have_statement_timeout= SHOW_OPTION_DISABLED;
 #endif
 
+#if !defined(EMBEDDED_LIBRARY)
+  if (separate_conn_handling_thread) {
+    for (auto ii : sockets_list) {
+      if (ii)
```

Comment:
yes deleting a null ptr is actually ok, but people still end up doing this. will remove.